packages/web-components/src/components/footer/language-selector.ts

```diff
@@ -0,0 +1,147 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2020, 2021
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { html, property, customElement, query } from 'lit-element';
+import ddsSettings from '@carbon/ibmdotcom-utilities/es/utilities/settings/settings.js';
+import BXComboBox from 'carbon-web-components/es/components/combo-box/combo-box.js';
+import BXComboBoxItem from 'carbon-web-components/es/components/combo-box/combo-box-item.js';
+import HostListener from 'carbon-web-components/es/globals/decorators/host-listener.js';
+import HostListenerMixin from 'carbon-web-components/es/globals/mixins/host-listener.js';
+import { FOOTER_SIZE } from './footer';
+import styles from './footer.scss';
+import { findIndex, forEach } from '../../globals/internal/collection-helpers';
+
+const { stablePrefix: ddsPrefix } = ddsSettings;
+
+/**
+ * Language selector
+ *
+ * @element dds-language-selector
+ */
+@customElement(`${ddsPrefix}-language-selector`)
+class DDSLanguageSelector extends HostListenerMixin(BXComboBox) {
+  /**
+   * Property that saves the last valid language to use on reset cases.
+   */
+
+  @property({ attribute: 'last-valid-lang' })
+  lastValidLang;
+
+  /**
+   * Size property to apply different styles.
+   */
+
+  @property({ attribute: 'footer-size' })
+  footerSize = FOOTER_SIZE.REGULAR;
```

Comment:
Does this cause any change in behavior of `<dds-language-selector>`?

Reply:
Not at the moment, but it's for the future `micro` styling

Reply:
Do you have some details? Wondering if we may have a better name of this property. Thanks!

Reply:
The React version has a `micro` version of the language selector, so just like in the `dds-locale-button` we're storing a `size` property to apply the proper styles for it. The reason why it's not `size` like in that component is that `size` is another property within `combo-box`, hence `footerSize` instead of just `size`

Reply:
I see thanks, do you have some screenshots for regular styling vs micro footer styling?

Reply:
Regular styles:
![Screen Shot 2021-01-07 at 5 38 15 PM](https://user-images.githubusercontent.com/24970122/103963952-32964400-510f-11eb-9038-c7cb67a5f27b.png)

Micro:
![Screen Shot 2021-01-07 at 5 38 22 PM](https://user-images.githubusercontent.com/24970122/103963956-34600780-510f-11eb-8571-88d32cc6fc73.png)

The background color, border, and position are the most prominent differences between the two. Here's the [React](https://ibmdotcom-react-canary.mybluemix.net/?path=/story/components-footer--micro-language-only)version for reference.

Reply:
Thanks @IgnacioBecerra! In this case I suggest (below code should be defined in `footer/defs.ts`):

```typescript
/**
 * The style scheme of language selector.
 */
export enum LANGUAGE_SELECTOR_STYLE_SCHEME {
  /**
   * Regular style scheme.
   */
  REGULAR = '',

  /**
   * The style scheme that's blendid into outer style scheme.
   * The primary use case is for used in micro footer.
   */
  BLENDED = 'blended',
}
```